compiler/src/iree/compiler/Codegen/Interfaces/BufferizationInterfaces.cpp

```diff
@@ -502,25 +502,60 @@ struct PackUnPackOpInterface
   }
 };
 
-/// Returns true if the value of a `storeOp` bufferizes to an equivalent
-/// DispatchTensorLoadOp result that bufferizes inplace.
-static bool isValueEquivalentToAnInplaceTensorLoadOp(
-    IREE::Flow::DispatchTensorStoreOp storeOp, Value candidate,
-    function_ref<bool(Value, Value)> equivalenceFn) {
-  auto loadOp = candidate.getDefiningOp<IREE::Flow::DispatchTensorLoadOp>();
-  if (!loadOp)
-    return false;
-  if (!equivalenceFn(loadOp.getSource(), storeOp.getTarget()))
+struct DispatchTensorLoadOpSubsetInterface
+    : public SubsetOpInterface::ExternalModel<
+          DispatchTensorLoadOpSubsetInterface,
+          IREE::Flow::DispatchTensorLoadOp> {
+  bool operatesOnEquivalentSubset(
+      Operation *op, SubsetOpInterface candidate,
+      function_ref<bool(Value, Value)> equivalenceFn) const {
+    // Returns true if the value of a `loadOp` bufferizes to an equivalent
+    // DispatchTensorStoreOp result that bufferizes inplace.
+    auto loadOp = cast<IREE::Flow::DispatchTensorLoadOp>(op);
+    auto storeOp = dyn_cast<IREE::Flow::DispatchTensorStoreOp>(op);
+    if (!storeOp)
+      return false;
+    return equivalenceFn(loadOp.getSource(), storeOp.getTarget());
+  }
+
+  bool operatesOnDisjointSubset(
+      Operation *op, SubsetOpInterface candidate,
+      function_ref<bool(Value, Value)> equivalenceFn) const {
+    // TODO: This is a new entry point and not clear it is correct.
     return false;
```

Comment:
We probably need @matthias-springer to weigh in on if it is correct

Reply:
He did on discord.

Reply:
cool cool! I checked discord, but somehow I missed it. Thanks!